src/components/CreateVmWizard/helpers.js

```diff
@@ -1,22 +1,67 @@
-import { createTemplateList } from '_/components/utils'
+import { createIsoList, createTemplateList } from '_/components/utils'
 import timezones from '_/components/utils/timezones.json'
 import { isWindows } from '_/helpers'
 
-const handleClusterIdChange = (clusterId, blankTemplateId, { templates, clusters, defaultValues, operatingSystems, defaultGeneralTimezone, defaultWindowsTimezone }) => {
+const handleClusterIdChange = (clusterId, { blankTemplateId, defaultValues, clusters, templates, operatingSystems, storageDomains, defaultGeneralTimezone, defaultWindowsTimezone }) => {
   let changes = {}
   const templateList = clusterId ? createTemplateList(templates, clusterId) : []
   changes.dataCenterId = clusterId ? clusters.getIn([clusterId, 'dataCenterId']) : undefined
   changes.clusterId = clusterId
-  changes.provisionSource =
-    (templateList.length === 1 && templateList[0].id === blankTemplateId)
-      ? 'iso'
-      : 'template'
-  changes.templateId = templateList.length === 1 ? blankTemplateId : undefined
-  changes = { ...changes, ...handleProvisionSourceChange(changes.provisionSource, defaultValues, { defaultGeneralTimezone, defaultWindowsTimezone, templates, operatingSystems }) }
+
+  if (clusterId === undefined) {
+    changes = { ...changes, ...handleProvisionSourceChange('template', { defaultValues, defaultGeneralTimezone, defaultWindowsTimezone, templates, operatingSystems }) }
+    return changes
+  }
+
+  if (templateList.length > 1) {
+    // cluster includes more than one non blank template
+    changes = {
+      ...changes,
+      ...handleProvisionSourceChange('template', { defaultValues, defaultGeneralTimezone, defaultWindowsTimezone, templates, operatingSystems }),
+    }
+    if (templateList.length === 2) {
+      // cluster includes only one non blank template
+      const selectedTemplate = templateList.find(t => t.id !== blankTemplateId)
+      changes = {
+        ...changes,
+        ...(selectedTemplate ? handleTemplateIdChange(selectedTemplate.id, defaultValues.optimizedFor, { templates, operatingSystems, defaultGeneralTimezone, defaultWindowsTimezone }) : {}),
+      }
+    }
+    return changes
+  }
+  if (templateList.length === 1 && templateList[0].id === blankTemplateId) {
+    // cluster includes only one template (blank) always has blank template
```

Comment:
Could you, please, change this comment a little bit? It's not 100% clear and gramatically correct, maybe you just wanted to write "cluster includes only one template (Blank)". Or maybe "cluster which includes only one template, has Blank template"?